### DISCUSSION TITLE
Issue with Metadata (Json) where we have multiples characters per image

### DISCUSSION THREAD

**Discussion by DevPratyush** [nftchef/art-engine#275] [Help + Q&A] [answered]
Thanks to @nftchef for creating this fork! I have been facing an issue and attempted many ways but it didn't get resolved.
Actually, I wanted to create a collection where we have 5 characters per image. We have traits for each character. Let me explain the structure of the folder.

I-Heroes
|--Hero 1
I----Hairs
l------Blackhair.png
l------Blondhair.png
l------Brownhair.png
I----Cloth
l------Blackcloth.png
l------Whitecloth.png
l------YellowCloth.png
I----Skin
l------Blackskin.png
l------Brownskin.png
l------Whiteskin.png
 
|--Hero 2
I----Hairs
l------Blackhair.png
l------Blondhair.png
l------Brownhair.png
I----Cloth
l------Blackcloth.png
l------Whitecloth.png
l------YellowCloth.png
I----Skin
l------Blackskin.png
l------Brownskin.png
l------Whiteskin.png

|--Hero 3
I----Hairs
l------Blackhair.png
l------Blondhair.png
l------Brownhair.png
I----Cloth
l------Blackcloth.png
l------Whitecloth.png
l------YellowCloth.png
I----Skin
l------Blackskin.png
l------Brownskin.png
l------Whiteskin.png

|--Hero 4
I----Hairs
l------Blackhair.png
l------Blondhair.png
l------Brownhair.png
I----Cloth
l------Blackcloth.png
l------Whitecloth.png
l------YellowCloth.png
I----Skin
l------Blackskin.png
l------Brownskin.png
l------Whiteskin.png

|--Hero 5
I----Hairs
l------Blackhair.png
l------Blondhair.png
l------Brownhair.png
I----Cloth
l------Blackcloth.png
l------Whitecloth.png
l------YellowCloth.png
I----Skin
l------Blackskin.png
l------Brownskin.png
l------Whiteskin.png

I-Background
I-Birds

Now, the generator works well for this but its metadata is not well structurally organized. It is how metadata is shown. 
![image](https://user-images.githubusercontent.com/105413515/168415502-63665a06-f899-4c32-b157-f4a2818501c6.png)
` {
      "trait_type": "Background",
      "value": "background 2"
    },
    {
      "trait_type": "Birds",
      "value": "birds 2"
    },
    {
      "trait_type": "Hero 1",
      "value": " Skin"
    },
    {
      "trait_type": " Cloth ",
      "value": "Black Cloth"
    },
    {
      "trait_type": " Hair",
      "value": "dark brown"
    }
    {
      "trait_type": "Hero 2",
      "value": " Skin"
    },
    {
      "trait_type": " Cloth ",
      "value": "Black Cloth"
    },
    {
      "trait_type": " Hair",
      "value": "dark brown"
    }`

There are three issues here - 
1) The structure is not good for multiple characters exist in each image 
2) It does not tell the value of Skin. See how it writes:   
"trait_type": "Hero 2",
    "value": " Skin"

3) The metadata does not tell Hair, Skin and Cloth are the traits of Heroes.   

Can anyone help me to make the metadata organized for this case?

**Comment by neilryanh** [comment, marked as answer]
1. Do you have weighted chances on each outcome (ie. Black Skin#30)?
2. Have you turned this feature on/off to see what you get: `const useRootTraitType = false;`?
3. Sometimes it's playing around with Folder/Images names until the metadata shows right. As long as the art is generating correctly, there's a lot of room in how you name your folders/images to work like you want.

**Comment by lazydevpro** [comment]
@DevPratyush can you share how to define this folder structure in config?

**Reply by nftchef** [reply]
@lazyDevPro reach out to me on Discord, or twitter, (usernames in readme), or start a discussion if you're having issues 🙇
